diff --git a/build/tools/roomservice.py b/build/tools/roomservice.py
index fd9ebd9a..acc35ece 100755
--- a/build/tools/roomservice.py
+++ b/build/tools/roomservice.py
@@ -206,7 +206,7 @@ def validate_repository(repo_name, target_path):
     repo_lower = repo_name.lower()
     target_lower = target_path.lower()
 
-    if not any(x in repo_lower for x in ('crdroidandroid', 'themuppets', 'lineageos')):
+    if not any(x in repo_lower for x in ('crdroidandroid', 'themuppets', 'lineageos', 'loukious')):
         return False, "Unsupported repository org"
 
     if 'lineageos' in repo_lower:
